:book: update RELEASE doc to provide the info about how the releases are done (#2662)
kubernetes-sigs/kubebuilder: RELEASE.md

- [kubebuilder-release-master-head-darwin-amd64.tar.gz](https://storage.googleapis.com/kubebuilder-release/kubebuilder-release-master-head-darwin-amd64.tar.gz)
- [kubebuilder-release-master-head-linux-amd64.tar.gz](https://storage.googleapis.com/kubebuilder-release/kubebuilder-release-master-head-linux-amd64.tar.gz)

## How the releases are configured

The releases occur in an account in the Google Cloud (See [here](https://console.cloud.google.com/cloud-build/builds?project=kubebuilder)) using Cloud Build.

### To build the Kubebuilder CLI binaries:

A trigger `build-kb-release` is configured to call [build/cloudbuild.yaml](build/cloudbuild.yaml).
This trigger will be executed when any new tag be published. 

> The tags must be build from the release branch (Currently, `release-3`).

Reviewer: ```suggestion
The tags must be built from the release branch (Currently, `release-3`).
```

Also, we have a trigger to generate snapshots builds from the master branch.
This trigger will call [build/cloudbuild_snapshot.yaml](build/cloudbuild_snapshot.yaml)

> when any change be performed on master.

Reviewer: ```suggestion
when any change needs to be performed on master.
```

> ### To build the Kubebuilder-tools: (Artefacts required to use ENV TEST)

Reviewer: ```suggestion
### To build the Kubebuilder-tools: (Artifacts required to use ENV TEST)
```

> Kubebuilder projects requires artefacts which are used to do test with ENV TEST (when we call `make test` target)

Reviewer: ```suggestion
Kubebuilder projects requires artifacts which are used to do test with ENV TEST (when we call `make test` target)
```

> These artefacts can be checked in the service page: https://storage.googleapis.com/kubebuilder-tools

Reviewer: ```suggestion
These artifacts can be checked in the service page: https://storage.googleapis.com/kubebuilder-tools
```

The build is made from the branch [tools-releases](https://github.com/kubernetes-sigs/kubebuilder/tree/tools-releases) and the trigger will call the `build/cloudbuild_tools.yaml` passing 
as argument the architecture and the SO that should be used, e.g:

> <img width="553" alt="Screenshot 2022-04-30 at 10 15 41" src="https://user-images.githubusercontent.com/7708031/166099666-ae9cd2df-73fe-47f6-a987-464f63df9a19.png">

Reviewer 2: Should this screenshot-png be part of the PR? I am not sure how images are handled in the docs. But could not find any [existing references](https://github.com/kubernetes-sigs/kubebuilder/search?q=user-images.githubusercontent.com).

Author: If we click on the view, we can check that. This doc is not used in the books. 
So that is fine. I mean, it works fine see:

<img width="481" alt="Screenshot 2022-05-09 at 13 26 56" src="https://user-images.githubusercontent.com/7708031/167409646-e624b768-f1c1-4377-8f70-5d242a98f14c.png">

<img width="1039" alt="Screenshot 2022-05-09 at 13 27 03" src="https://user-images.githubusercontent.com/7708031/167409658-a1516b6a-56d1-4ca5-8440-d0ddce32464a.png">

For further information see the [README](https://github.com/kubernetes-sigs/kubebuilder/blob/tools-releases/README.md).

### To build the `kube-rbac-proxy` images:

> These images are build from the project [brancz/kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy).

Reviewer: ```suggestion
These images are built from the project [brancz/kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy).
```

The projects built with Kubebuilder creates a side container with `kube-rbac-proxy` to protect the Manager.

These images are can be checked in the consolse, see [here](https://console.cloud.google.com/gcr/images/kubebuilder/GLOBAL/kube-rbac-proxy).

The project `kube-rbac-proxy` is in the process to be donated to the k8s org. However, it is going on for a long time and then,
we have no ETA for that to occur. When that occurs we can automate this process. But until there we need to generate these images
by bumping the versions/tags released by `kube-rbac-proxy` on the branch 
[kube-rbac-proxy-releases](https://github.com/kubernetes-sigs/kubebuilder/tree/kube-rbac-proxy-releases)
then the `build/cloudbuild_kube-rbac-proxy.yaml` will generate the images.

To check an example, see the pull request [#2578](https://github.com/kubernetes-sigs/kubebuilder/pull/2578).

**Note**: we cannot use the images produced by the project `kube-rbac-proxy` because we need to ensure
to Kubebuilder users that these images will be available.

### To build the `gcr.io/kubebuilder/pr-verifier` images:

> These images are used to verify the PR title and description. They are build from [kubernetes-sigs/kubebuilder-release-tools](https://github.com/kubernetes-sigs/kubebuilder-release-tools/).

Reviewer: ```suggestion
These images are used to verify the PR title and description. They are built from [kubernetes-sigs/kubebuilder-release-tools](https://github.com/kubernetes-sigs/kubebuilder-release-tools/).
```

In Kubebuilder, we have been using this project via the GitHub action [.github/workflows/verify.yml](.github/workflows/verify.yml)
and not the image, see:

```yaml
  verify:
    name: Verify PR contents
    runs-on: ubuntu-latest
    steps:
    - name: Verifier action
      id: verifier
      uses: kubernetes-sigs/kubebuilder-release-tools@v0.1.1
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
```

> However, the image should still be build and maintained since others projects under the org might be using them. 

Reviewer: ```suggestion
However, the image should still be built and maintained since other projects under the org might be using them. 
```

> However, the image should still be built and maintained since other projects under the org might be using them. 

Reviewer 3: Does this step need to be done for every Kubebuilder release? 

I didn't see any steps for the proper way to build and verify that the pr-verifier images are built and in the right place. Is this an automated process that only occurs when changes are made to the pr-verifier?

Author: No. This step does need to be done for every KB release. This trigger is done for the repo: kubernetes-sigs/kubebuilder-release-tools and not in the Kubebuilder repo.

Reviewer 3: Since that is a separate repo, should release steps for that image be provided in that repo instead of the KB release steps?

How does this relate to a KB release other than that the pr-verifier action is used to verify KB PRs?

Author: Becuase it is maintained by us. 
See that it is kubebuilder-tools-release and is under Kubebuilder project when we check the Google cloud triggers. So, I just want to make clear the info. 

Any extra information should be checked in its repo. We can improve it afterwords. For now, my goal is we have the info at least in case someone need to check it out.
